plugin/ochttp/client_trace.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ochttp
+
+import (
+	"crypto/tls"
+	"net/http/httptrace"
+	"strings"
+
+	"go.opencensus.io/trace"
+)
+
+// ClientTrace implements the annotation of all available hooks for
+// httptrace.ClientTrace.
+type ClientTrace struct {
+	*trace.Span
+}
+
+// ClientTracer returns a httptrace.ClientTrace instance which instruments all
+// emitted httptrace events on the provided Span.
+func ClientTracer(s *trace.Span) *httptrace.ClientTrace {
```

Comment:
The name here doesn't add much. Maybe `NewSpanAnnotator`?

Reply:
ok